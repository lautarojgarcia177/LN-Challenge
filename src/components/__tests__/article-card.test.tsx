import { render, screen } from "@testing-library/react";
import ArticleCard from "../article-card";

describe("ArticleCard", () => {
  const defaultProps = {
    imageUrl: "https://example.com/image.jpg",
    title: "Titulo de ejemplo",
    date: "6 de diciembre de 2019",
    subtitle: "Subtitulo de ejemplo",
  };

  it("coincide con la snapshot", () => {
    const { container } = render(<ArticleCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renderiza la article card con una imagen de placeholder cuando la imageUrl no es provista", () => {
    const { imageUrl, ...propsWithoutImageUrl } = defaultProps;
    render(<ArticleCard {...propsWithoutImageUrl} />);

    const imageElement = screen.getByAltText("");
    expect(imageElement).toHaveAttribute("src", "/images/placeholder.webp");
  });

  it("renderiza la article card sin ningun subtitulo cuando este no es provisto", () => {
    const { subtitle, ...propsWithoutSubtitle } = defaultProps;
    render(<ArticleCard {...propsWithoutSubtitle} />);

    const titleElement = screen.getByText(/Titulo de ejemplo/i);
    const dateElement = screen.getByText(/6 de diciembre de 2019/i);
    const subtitleElement = screen.queryByText(/Subtitulo de ejemplo/i);

    expect(titleElement).toBeInTheDocument();
    expect(dateElement).toBeInTheDocument();
    expect(subtitleElement).not.toBeInTheDocument();
  });
});
