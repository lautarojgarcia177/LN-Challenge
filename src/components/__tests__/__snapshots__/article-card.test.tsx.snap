// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard coincide con la snapshot 1`] = `
<div>
  <article
    class="mod-caja-nota w-100-mobile"
  >
    <section
      class="cont-figure"
    >
      <a
        class="figure"
        href=""
      >
        <picture
          class="content-pic picture"
        >
          <img
            alt=""
            class="content-img"
            src="https://example.com/image.jpg"
          />
        </picture>
      </a>
    </section>
    <div
      class="mod-caja-nota__descrip"
    >
      <h2
        class="com-title-acu"
      >
        <a
          href=""
        >
          <b>
            Titulo de ejemplo
            .
          </b>
          Â 
          Subtitulo de ejemplo
        </a>
      </h2>
      <h4
        class="com-date"
      >
        6 de diciembre de 2019
      </h4>
    </div>
  </article>
</div>
`;
